common/src/main/java/com/microsoft/identity/common/internal/authorities/AzureActiveDirectoryB2CAuthority.java

```diff
@@ -93,4 +93,17 @@ public OAuth2Strategy createOAuth2Strategy(@NonNull final OAuth2StrategyParamete
         MicrosoftStsOAuth2Configuration config = createOAuth2Configuration();
         return new MicrosoftStsOAuth2Strategy(config, parameters);
     }
+
+    /**
+     * This method attempts to split the Authority URI
+     * and return the last item, which is the policy name.
+     * The authority format for Azure AD B2C is: https://{azureADB2CHostname}/tfp/{tenant}/{policyName}
+     *
+     * @return a String with the Policy name
+     */
+    public String getB2CPolicyName(){
+        final String authorityUriString = this.getAuthorityUri().toString();
```

Comment:
>his.getAuthorityUri().toString(); [](http://example.com/codeflow?start=43&length=33)

Why not use "mAuthorityUrl" directly?

We can also expose the policyName as a member variable for this class and set it in the constructor.

Reply:
using mAuthorityUrl